### PR TITLE
fix failed assert on generic fn opaque return type

### LINKED ISSUE #4587: zeroes(@OpaqueType()) triggers assertion in compiler.
```zig
var widely_regarded_as_a_bad_move = zeroes(@OpaqueType());
```
```
Assertion failed at D:\a\1\s\src\analyze.cpp:1003 in get_fn_type. This is a bug in the Zig compiler.
Unable to dump stack trace: debug info stripped
```

While that code is total nonsense, it shouldn't crash the compiler. 0.5.0+68dbba212

### PR DESCRIPTION
**Pull request by xackus**
fixes #4587

### CODE REVIEW

**Comment by andrewrk** [comment]
Would you mind rebasing against master so we can check if those tests are passing?

**Comment by xackus** [comment]
@andrewrk tests have passed

File: test/compile_errors.zig [hunk 1]
```diff
@@ -6539,8 +6539,22 @@ pub fn addCases(cases: *tests.CompileErrorContext) void {
         \\    return error.InvalidValue;
         \\}
     , &[_][]const u8{
-        "tmp.zig:2:18: error: opaque return type 'FooType' not allowed",
-        "tmp.zig:1:1: note: declared here",
+        "tmp.zig:2:18: error: return type 'FooType' not allowed",
```
Comment (andrewrk) [inline test/compile_errors.zig]: This error message is less helpful now. Can we have the old message?

Comment (xackus) [reply test/compile_errors.zig]: I found the right function: `type_id_name`. Done.

**Comment by andrewrk** [comment]
Nice work!

**Pull request merged**